// ngx-pipes/esm2020/src/ng-pipes/pipes/date/index.mjs
import { NgModule } from '@angular/core';
import { TimeAgoPipe } from './time-ago';
import * as i0 from "@angular/core";
export const DATE_PIPES = [TimeAgoPipe];
export class NgDatePipesModule {
}
NgDatePipesModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NgDatePipesModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgDatePipesModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: NgDatePipesModule, declarations: [TimeAgoPipe], exports: [TimeAgoPipe] });
NgDatePipesModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NgDatePipesModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NgDatePipesModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: DATE_PIPES,
                    imports: [],
                    exports: DATE_PIPES,
                }]
        }] });
export { TimeAgoPipe } from './time-ago';
